<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>yt-dlp-gui-qt</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLineEdit" name="txtUrl">
        <property name="placeholderText">
         <string>Video URL</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="btnProcess">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="styleSheet">
         <string notr="true">padding-top: 4px;
padding-bottom: 4px;
padding-left: 12px;
padding-right: 12px;</string>
        </property>
        <property name="text">
         <string>Process</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLineEdit" name="txtVideoTitle">
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="placeholderText">
         <string>Video title</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="1" column="0">
         <widget class="QLabel" name="label_2">
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Audio format:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="cmbVideoFormats">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Video format:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="btnDownloadVideo">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="maximumSize">
           <size>
            <width>30</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Download Video</string>
          </property>
          <property name="styleSheet">
           <string notr="true">padding-top: 5px;
padding-bottom: 5px;</string>
          </property>
          <property name="text">
           <string>D</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QComboBox" name="cmbAudioFormats">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QPushButton" name="btnDownloadAudio">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="maximumSize">
           <size>
            <width>30</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Download Audio</string>
          </property>
          <property name="styleSheet">
           <string notr="true">padding-top: 5px;
padding-bottom: 5px;</string>
          </property>
          <property name="text">
           <string>D</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="4" column="0">
       <layout class="QFormLayout" name="formLayout"/>
      </item>
      <item row="1" column="0">
       <widget class="QPlainTextEdit" name="txtVideoDescription">
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="placeholderText">
         <string>Video description</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="chkQuality">
        <property name="text">
         <string>Automatic best quality</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1" rowspan="5">
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QFrame" name="frame">
          <property name="minimumSize">
           <size>
            <width>300</width>
            <height>168</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>300</width>
            <height>168</height>
           </size>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <widget class="QLabel" name="imgThumbnail">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>300</width>
             <height>168</height>
            </rect>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
          <widget class="QLabel" name="lblDuration">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="geometry">
            <rect>
             <x>2</x>
             <y>142</y>
             <width>49</width>
             <height>21</height>
            </rect>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="font">
            <font>
             <pointsize>10</pointsize>
             <bold>true</bold>
            </font>
           </property>
           <property name="toolTipDuration">
            <number>-1</number>
           </property>
           <property name="layoutDirection">
            <enum>Qt::LeftToRight</enum>
           </property>
           <property name="styleSheet">
            <string notr="true">background-color: rgba(38, 46, 56, 0.8);
border-top-left-radius: 5px;
padding-left: 6px;
padding-right: 6px;</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="scaledContents">
            <bool>true</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignJustify|Qt::AlignVCenter</set>
           </property>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="chkSaveThumbnail">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Save thumbnail</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="1" column="2">
       <widget class="QPushButton" name="btnDownload">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="styleSheet">
         <string notr="true">padding-top: 4px;
padding-bottom: 4px;
padding-left: 12px;
padding-right: 12px;</string>
        </property>
        <property name="text">
         <string>Download</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLineEdit" name="txtOutputFilename">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="btnBrowse">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="styleSheet">
         <string notr="true">padding-top: 4px;
padding-bottom: 4px;
padding-left: 12px;
padding-right: 12px;</string>
        </property>
        <property name="text">
         <string>Browse</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionOptions"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuTools"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionOptions">
   <property name="text">
    <string>Options</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>btnProcess</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>process_button_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>749</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>796</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtUrl</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>url_text_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>363</x>
     <y>44</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnBrowse</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>browse_button_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>671</x>
     <y>378</y>
    </hint>
    <hint type="destinationlabel">
     <x>671</x>
     <y>416</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnDownload</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>download_button_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>739</x>
     <y>378</y>
    </hint>
    <hint type="destinationlabel">
     <x>739</x>
     <y>405</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkQuality</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>quality_toggled()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>121</x>
     <y>267</y>
    </hint>
    <hint type="destinationlabel">
     <x>162</x>
     <y>409</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmbVideoFormats</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>video_format_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>302</y>
    </hint>
    <hint type="destinationlabel">
     <x>322</x>
     <y>407</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnDownloadVideo</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>download_video_button_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>466</x>
     <y>297</y>
    </hint>
    <hint type="destinationlabel">
     <x>477</x>
     <y>423</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnDownloadAudio</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>download_audio_button_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>471</x>
     <y>337</y>
    </hint>
    <hint type="destinationlabel">
     <x>471</x>
     <y>417</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>process_button_clicked()</slot>
  <slot>url_text_changed()</slot>
  <slot>browse_button_clicked()</slot>
  <slot>download_button_clicked()</slot>
  <slot>video_format_changed()</slot>
  <slot>quality_toggled()</slot>
  <slot>download_video_button_clicked()</slot>
  <slot>download_audio_button_clicked()</slot>
 </slots>
</ui>
